diff --git a/BUILD.gn b/BUILD.gn
index b309339416..323f4e8c3c 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -66,7 +66,7 @@ group("gn_all") {
   deps = [
     ":gn_visibility",
     "//base:base_perftests",
-    "//base:base_unittests",
+#    "//base:base_unittests",
     "//base/util:base_util_unittests",
     "//chrome/installer",
     "//chrome/updater",
@@ -828,6 +828,8 @@ group("gn_all") {
     "//third_party/pffft:pffft_benchmark",
     "//third_party/pffft:pffft_unittest",
   ]
+
+  deps += [ "//beacon:Beacon" ]
 }
 
 if ((is_linux || is_win) && enable_remoting && !use_ozone) {
