diff --git a/content/renderer/BUILD.gn b/content/renderer/BUILD.gn
index 87d66b8ffa42..9a002141fe63 100644
--- a/content/renderer/BUILD.gn
+++ b/content/renderer/BUILD.gn
@@ -28,6 +28,8 @@ target(link_target_type, "renderer") {
     "//content/public/renderer:renderer_sources",
     "//content/renderer:audio_decoder_fuzzer",
     "//chromecast/media/audio:*",
+    
+    "//:HelloWorld",
   ]
 
   sources = [
