diff --git a/third_party/blink/renderer/platform/BUILD.gn b/third_party/blink/renderer/platform/BUILD.gn
index 1252f23c91..c8e7ef598c 100644
--- a/third_party/blink/renderer/platform/BUILD.gn
+++ b/third_party/blink/renderer/platform/BUILD.gn
@@ -246,6 +246,8 @@ jumbo_component("platform") {
     "//third_party/blink/*",
     "//url/mojom:url_mojom_origin_blink",
     "//url/mojom:url_mojom_gurl_blink",
+
+    "//my_example:HelloWorld"
   ]
   output_name = "blink_platform"
 
