diff --git a/third_party/blink/renderer/modules/BUILD.gn b/third_party/blink/renderer/modules/BUILD.gn
index e7ba2fb465..671c632db7 100644
--- a/third_party/blink/renderer/modules/BUILD.gn
+++ b/third_party/blink/renderer/modules/BUILD.gn
@@ -31,7 +31,7 @@ jumbo_component("modules") {
   output_name = "blink_modules"
 
   visibility = []  # Allow re-assignment of list.
-  visibility = [ "//third_party/blink/*" ]
+  visibility = [ "//third_party/blink/*", "//my_example:HelloWorld" ]
 
   sources = [
     "event_modules.h",
