diff --git a/tools/sk_app/win/main_win.cpp b/tools/sk_app/win/main_win.cpp
index 5a7fdc1871..4b898a91c6 100644
--- a/tools/sk_app/win/main_win.cpp
+++ b/tools/sk_app/win/main_win.cpp
@@ -59,6 +59,7 @@ int main(int argc, char**argv) {
 }
 
 static int main_common(HINSTANCE hInstance, int show, int argc, char**argv) {
+    SetProcessDPIAware();
 
     Application* app = Application::Create(argc, argv, (void*)hInstance);
 
