diff --git a/tools/sk_app/win/Window_win.cpp b/tools/sk_app/win/Window_win.cpp
index 43ed343a30..0f29ab70cc 100644
--- a/tools/sk_app/win/Window_win.cpp
+++ b/tools/sk_app/win/Window_win.cpp
@@ -77,7 +77,7 @@ bool Window_win::init(HINSTANCE hInstance) {
         wcex.hInstance = fHInstance;
         wcex.hIcon = LoadIcon(fHInstance, (LPCTSTR)IDI_WINLOGO);
         wcex.hCursor = LoadCursor(nullptr, IDC_ARROW);
-        wcex.hbrBackground = (HBRUSH)(COLOR_WINDOW + 1);
+        wcex.hbrBackground = NULL;
         wcex.lpszMenuName = nullptr;
         wcex.lpszClassName = gSZWindowClass;
         wcex.hIconSm = LoadIcon(fHInstance, (LPCTSTR)IDI_WINLOGO);
@@ -109,7 +109,8 @@ bool Window_win::init(HINSTANCE hInstance) {
     */
  //   gIsFullscreen = fullscreen;
 
-    fHWnd = CreateWindow(gSZWindowClass, nullptr, WS_OVERLAPPEDWINDOW,
+    fHWnd = CreateWindowEx(WS_EX_OVERLAPPEDWINDOW | WS_EX_COMPOSITED, gSZWindowClass, nullptr,
+                           WS_OVERLAPPEDWINDOW,
                          gWindowX, gWindowY, gWindowWidth, gWindowHeight,
                          nullptr, nullptr, fHInstance, nullptr);
     if (!fHWnd)
@@ -233,6 +234,16 @@ LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
             // disable/enable rendering here, depending on wParam != WA_INACTIVE
             break;
 
+        case WM_ENTERSIZEMOVE:
+            window->onBeginResizing();
+            eventHandled = true;
+            break;
+
+        case WM_EXITSIZEMOVE:
+            window->onEndResizing();
+            eventHandled = true;
+            break;
+
         case WM_SIZE:
             window->onResize(LOWORD(lParam), HIWORD(lParam));
             eventHandled = true;
